configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -0,0 +1,386 @@
+            out.println("@ConfigurationProperties(prefix = \"" +
+                    prefix + "\")");
+            out.println("public class " + simpleClassName + " {");
+
+            fields.forEach(f -> writeField(out, f));
+
+            writePostConstruct(out, fields);
+
+            out.println("}");
+        }
+    }
+
+    private String getImportsAsString(String className) {
+        StringBuilder value = new StringBuilder();
+        CompilationUnit cu = getCompilationUnit(className);
+        if (cu != null) {
+            cu.accept(new VoidVisitorAdapter<StringBuilder>() {
+
+                @Override
+                public void visit(ImportDeclaration fieldDeclaration,
+                                  StringBuilder arg) {
+                    super.visit(fieldDeclaration, arg);
+                    value.append(fieldDeclaration.toString());
+                }
+
+            }, value);
+        }
+        return value.toString();
+    }
+
+    private void writePostConstruct(PrintWriter out,
+                                    List<FieldDescriptor> fields) {
+        out.println(" @PostConstruct");
+        out.println(" public void transferValuesToSystem() {");
+        fields.forEach(f -> {
+            out.println("  if (" + f.fieldName + " != null) " +
+                    "System.setProperty(\"vaadin." + f.formerPropertyName
+                    + "\", " + f.fieldName + ".toString());");
+        });
+        out.println(" }");
+    }
+
+    private void writeField(PrintWriter out, FieldDescriptor field) {
+        out.println();
+
+        // write comment
+        String comment = field.comments;
+        if (comment != null && !"".equals(comment)) {
+            out.println("  /**");
+            Arrays.stream(comment.split("\n")).
+                    forEach(l -> out.println("  * " + l));
+            out.println("  */");
+        }
+        // write field
+        out.print("  private " + field.type + " " + field.fieldName);
+        if (field.defaultValueSetter != null)
+            out.print(" = " + field.defaultValueSetter);
+        out.println(";");
+
+        out.println();
+        // write getter
+        out.println("  public " + field.type + " get" +
+                capitalize(field.fieldName) + "() {");
+        out.println("    return " + field.fieldName + ";");
+        out.println("  };");
+        out.println();
+        // write setter
+        out.println("  public void set" + capitalize(field.fieldName)
+                + "(" + field.type + " " + field.fieldName + ") {");
+        out.println("    this." + field.fieldName + " = " +
+                field.fieldName + ";");
+        out.println("  };");
+
+    }
+
+    private String toCamelCase(String value, boolean firstUppercase) {
+        if (value == null) return null;
+        StringBuffer camelized = new StringBuffer();
+        Arrays.stream(value
+                .replaceAll("([A-Z])", "\\-$1")
+                .split("[\\.\\-_)]"))
```

Comment:
well, the generated configuration class plains all of the property names. Why? because we have properties like `pnpm.enable` or `devmode.live-reload.enabled` which use dots though they are not intended to be complex objects. 

In fact, all of those values are read as simple String values from the vaadin servlet, so I understood it made no sense to generate a complex classes structure for those cases.

All this also enables using `pnpm-enable` as much as `pnpm.enable`, which was something expected.

Also, `toCamelCase` is intended for generating a method name, so dots must be converted to something to generate a valid method name.

Reply:
We are talking about totally different things.

There is no sense to generate complex structures: yes.

You are saying that "pnpm.enable" property in `application.properties` is the same as `pnpm-enable`.
This is UNEXPECTED to me (since I thought that Spring at least uses prefix `"vaadin."  for a namespace  which means that `"vaadin.foo"` can't be the same as `"vaadin-foo"` because they are in different namespaces.
May be I'm wrong and they are treated in the same way by Spring: I would like to see the confirmation of this.